src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.BaseNameProvider.cs

```diff
@@ -0,0 +1,146 @@
+                provider.nameParts = null;
+                s_providerPool.Free(provider);
+            }
+
+            public object GetTypeFromDefinition(MetadataReader reader, TypeDefinitionHandle handle, byte rawTypeKind)
+            {
+                // Add the namespace and name of this type definition.  But only do this
+                // for the first type we hit.  Any further types will be things like
+                // type arguments, as we do not want to add those to the list.
+                if (nameParts.Count == 0)
+                {
+                    var typeDefinition = reader.GetTypeDefinition(handle);
+                    var declaringType = typeDefinition.GetDeclaringType();
+                    if (declaringType.IsNil)
+                    {
+                        // Not a nested type, just add the containing namespace.
+                        AddNamespaceParts(reader, typeDefinition.NamespaceDefinition);
+                    }
+                    else
+                    {
+                        // We're a nested type, recurse and add the type we're declared in.
+                        // It will handle adding the namespace properly.
+                        GetTypeFromDefinition(reader, declaringType, rawTypeKind);
+                    }
+
+                    // Now add the simple name of the type itself.
+                    nameParts.Add(GetMetadataNameWithoutBackticks(reader, typeDefinition.Name));
+                }
+                return null;
+            }
+
+            private void AddNamespaceParts(
+                MetadataReader reader, NamespaceDefinitionHandle namespaceHandle)
+            {
+                if (namespaceHandle.IsNil)
+                {
+                    return;
+                }
+
+                var namespaceDefinition = reader.GetNamespaceDefinition(namespaceHandle);
+                AddNamespaceParts(reader, namespaceDefinition.Parent);
+                nameParts.Add(reader.GetString(namespaceDefinition.Name));
+            }
+
+            public object GetTypeFromReference(MetadataReader reader, TypeReferenceHandle handle, byte rawTypeKind)
+            {
+                // Add the namespace and name of this type reference.  But only do this
+                // for the first type we hit.  Any further types will be things like
+                // type arguments, as we do not want to add those to the list.
+                if (nameParts.Count == 0)
+                {
+                    var typeReference = reader.GetTypeReference(handle);
+                    var namespaceString = reader.GetString(typeReference.Namespace);
+
+                    // NOTE(cyrusn): Unfortunately, we are forced to allocate here
+                    // no matter what.  The metadata reader API gives us no way to
+                    // just get the component namespace parts for a namespace reference.
+                    nameParts.AddRange(namespaceString.Split(s_dotSeparator));
```

Comment:
Not currently. Please file an issue on [http://github.com/dotnet/corefx](http://github.com/dotnet/corefx)

---
In reply to: [75917758](https://github.com/dotnet/roslyn/pull/13266#discussion_r75917758) [](ancestors = 75917758)